capstone-frontend/src/components/Home.jsx

```diff
@@ -0,0 +1,309 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { CardDeck, Tabs, Tab, Card, Spinner, Button } from 'react-bootstrap';
+import ClubGridItem from './ClubGridItem.jsx';
+import { BookDescriptionOverlay } from "./BookDescriptionOverlay";
+import CreateList from './CreateList';
+import '../App.css';
+import '../styles/Home.css';
+
+export class Home extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      clubsOwned: [],
+      clubsMember: [],
+      myBookListsMap: new Map(),
+      sharedBookListsMap: new Map(),
+      key: 'Clubs - Owner', // For Bootstrap Tabs
+      screenWidth: window.outerWidth,
+      fetchingClubsOwned: true, // For Spinner
+      fetchingClubsMember: true, // For Spinner
+      fetchingBookListsOwned: true, // For Spinner
+      fetchingBookListsShared: true, // For Spinner
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("resize", this.resize);
+    this.fetchClubs('own');
+    this.fetchClubs('shared');
+    this.fetchBookLists('own');
+    this.fetchBookLists('shared');
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  resize = () => {
+    this.setState({ screenWidth: window.outerWidth });
+
+    this.fetchBookLists('own');
```

Comment:
It does a re-fetch because depending on the screen size, it will only fetch either 1, 3 or 5 books. This could probably be optimize so it doesn't do a fetch, but I think I should do that in another PR.

Reply:
We have to re-fetch booklists because depending on the screen size of the device it will only fetch 1, 3, or 5 books, making it compatible with multiple devices.